main.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/structs"
+	"github.com/iancoleman/strcase"
+	"github.com/sirupsen/logrus"
+	"github.com/vshn/appcat-cmd/internal/defaults"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+func init() {
+	logrus.SetOutput(os.Stderr)
+}
+
+func main() {
+	code := Main(os.Args, os.Stdin, os.Stdout)
+	os.Exit(code)
+}
+
+func Main(args []string, in io.Reader, out io.Writer) int {
+	plainArgs := args
+	service, err := findServiceType(plainArgs[1])
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	plainArgs, err = cleanInputArguments(plainArgs)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	parameters := parseArgs(plainArgs)
+
+	_, err = decorateType(service, parameters)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	err = writeYAML(service, os.Stdout)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	return 0
+}
+
+func writeYAML(service runtime.Object, out io.Writer) error {
+	yp := printers.YAMLPrinter{}
+	err := yp.PrintObj(service, out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Split input parameter of type "--foo=bar" to ["--foo", "bar"]
+func splitArguments(argument string) ([2]string, error) {
+	if strings.Count(argument, "=") > 1 {
```

Comment:
I don't like this because this makes it impossible to use "=" in values, and I see no reason to disallow that.

I suggest just splitting the string at the first "=" and leaving everything after that intact. The strings.Split function is probably not the right one to use for that, according to the strings docs "strings.Cut" might be the way to go.

Reply:
Done